tests/test_common/test_keymap_key.cpp

```diff
@@ -15,16 +15,27 @@
  */
 
 #include "test_keymap_key.hpp"
+#include <cstdint>
+#include "matrix.h"
 #include "test_logger.hpp"
 #include "gtest/gtest-message.h"
 #include "gtest/gtest.h"
+#include "timer.h"
 
 void KeymapKey::press() {
-    test_logger.trace() << "Key pressed:  (" << +this->position.col << "," << +this->position.row << ")" << std::endl;
+    EXPECT_FALSE(matrix_is_on(position.row, position.col)) << "Tried to press key (" << +this->position.col << "," << +this->position.row << ") "
+                                                           << "that wasn't pressed before! Check the test code." << std::endl;
```

Comment:
```suggestion
                                                           << "that was already pressed! Check the test code." << std::endl;
```
Decoupling the tests sounds like a good idea to me, but I don't think it will impact the tapdance tests that I am trying to write.

Reply:
What would help you developing the tap dance unit-tests?

Reply:
Some more time; I'm making progress and already found a few bugs.

Reply:
Alright, I'm not trying to push anything. Just asking if you perhaps need more facilities that would help you.